### ISSUE TITLE
Feature request: Custom color themes

### ISSUE THREAD

**Issue by holmanb** [justjanne/powerline-go#392]
**Is your feature request related to a problem? Please describe.**
I cannot customize the color palette.

**Describe the solution you'd like**
I'd love to be able to define on the commandline or in a configuration file which colors to use.

**Describe alternatives you've considered**
I've tried default, low-contrast, gruvbox, solarized-dark16, solarized-light16. I'd like a low-contrast theme which goes well with my preferred dark terminal. The default is the least offensive to me, but I'd like to have a default which I can fully customize to meet my needs.

**Comment by SKrPl** [comment]
Yes, this is much needed, I recently started using [catppuccin](https://catppuccin.com/) and I wanted to use the same in powerline-go, but can't at the moment.

**Comment by holmanb** [comment]
> Yes, this is much needed, I recently started using [catppuccin](https://catppuccin.com/) and I wanted to use the same in powerline-go, but can't at the moment.

Agreed, catpuccin is my preferred theme and I would love to see powerline-go match this color scheme.

**Comment by Justin08784** [comment]
This would be greatly appreciated!

**Comment by gentoomaniac** [comment]
That is actually supported.
You can specify a path in `theme` ending with `.json` and the [code](https://github.com/justjanne/powerline-go/blob/059f7f230760f8800307b3ae632c6cc6ca3f81d4/main.go#L222-L234) tries to load it.
A sample of the config file is [here](https://github.com/justjanne/powerline-go/blob/main/themes/default.json)
As far as I understand the code you can also specify it in `~/.config/powerline-go/config.json` in the `Themes` [src](https://github.com/justjanne/powerline-go/blob/059f7f230760f8800307b3ae632c6cc6ca3f81d4/config.go#L56) map, key is the theme name and the linked structure would be the value.

Haven't tested this myself though.
